test/linux_target_test.go

```diff
@@ -666,6 +669,517 @@ echo "$BAR" > bar.txt
+
+			dt, ok := res.Metadata[exptypes.ExporterImageConfigKey]
+			assert.Assert(t, ok, "result metadata should contain an image config: available metadata: %s", strings.Join(maps.Keys(res.Metadata), ", "))
+
+			var cfg dalec.DockerImageSpec
+			assert.Assert(t, json.Unmarshal(dt, &cfg))
+			assert.Check(t, cfg.Created.After(beforeBuild))
+			assert.Check(t, cfg.Created.Before(time.Now()))
+
+			// Make sure the test framework was actually executed by the build target.
+			// This appends a test case so that is expected to fail and as such cause the build to fail.
+			spec.Tests = append(spec.Tests, &dalec.TestSpec{
+				Name: "Test framework should be executed",
+				Steps: []dalec.TestStep{
+					{Command: "/bin/sh -c 'echo this command should fail; exit 42'"},
+				},
+			})
+
+			// update the spec in the solve request
+			withSpec(ctx, t, &spec)(&newSolveRequestConfig{req: &sr})
+
+			_, solveErr := gwc.Solve(ctx, sr)
+			if solveErr == nil {
+				t.Fatal("expected test spec to run with error but got none")
+			}
+
+			// Map Docker to systemd architecture. Some (e.g. arm64) are the
+			// same and are covered by the default case.
+			var systemdArch string
+			switch cfg.Platform.Architecture {
+			case "amd64":
+				systemdArch = "x86-64"
+			default:
+				systemdArch = cfg.Platform.Architecture
+			}
+
+			ref, refErr := res.SingleRef()
+			if refErr != nil {
+				t.Fatal(refErr)
+			}
+
+			expectedPath := fmt.Sprintf("/test-sysext-build-v0.0.1-1-%s.raw", systemdArch)
+			_, statErr := ref.StatFile(ctx, gwclient.StatRequest{Path: expectedPath})
+			if statErr != nil {
+				t.Fatalf("expected sysext image not found: %v", statErr)
+			}
+
+			sr = newSolveRequest(withBuildTarget(testConfig.Target.Worker), withSpec(ctx, t, nil))
+			worker := testConfig.Worker.SysextWorker(reqToState(ctx, gwc, sr, t))
+
+			pc := dalec.Platform(&cfg.Platform)
+			var opts []llb.ConstraintsOpt
+			opts = append(opts, pc)
+
+			state, stateErr := ref.ToState()
+			if stateErr != nil {
+				t.Fatal(stateErr)
+			}
+
+			output := worker.Run(
+				llb.Args([]string{"fsck.erofs", "--extract=/output", "/input" + expectedPath}),
+				llb.AddMount("/input", state, llb.Readonly),
+				dalec.WithConstraints(opts...),
+			).AddMount("/output", llb.Scratch())
+
+			def, defErr := output.Marshal(ctx, pc)
+			if defErr != nil {
+				t.Fatalf("error marshalling llb: %v", defErr)
+			}
+
+			res, resErr := gwc.Solve(ctx, gwclient.SolveRequest{
+				Definition: def.ToPB(),
+			})
+			if resErr != nil {
+				t.Fatal(resErr)
+			}
+
+			ref, refErr = res.SingleRef()
+			if refErr != nil {
+				t.Fatal(refErr)
+			}
+			if evalErr := ref.Evaluate(ctx); evalErr != nil {
+				t.Fatalf("error extracting sysext: %v", evalErr)
+			}
+
+			for _, file := range []string{"/usr/bin/jq", "/usr/bin/zsh"} {
+				_, statErr = ref.StatFile(ctx, gwclient.StatRequest{Path: file})
+				if statErr != nil {
+					t.Fatalf("expected file in sysext not found: %v", statErr)
+				}
+			}
+
```

Comment:
Can you add a check showing that dependencies were not included? `jq` should have a predictable dependency for which you can do an absence check.